src/test/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/resource/PrisonIncentiveLevelResourceTest.kt

```diff
@@ -1216,6 +1216,297 @@ class PrisonIncentiveLevelResourceTest : IncentiveLevelResourceTestBase() {
       assertNoAuditMessageSent()
     }
 
+    @Test
+    fun `does nothing when trying to deactivate all prison incentive levels if none exist`() {
+      webTestClient.delete()
+        .uri("/incentive/prison-levels/BAI")
+        .withLocalAuthorisation()
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isOk
+        .expectBody().json(
+          // language=json
+          """
+          []
+          """,
+          true,
+        )
+
+      runBlocking {
+        assertThat(prisonIncentiveLevelRepository.count()).isZero
+      }
+
+      assertNoDomainEventSent()
+      assertNoAuditMessageSent()
+    }
+
+    @Test
+    fun `deactivates all prison incentive levels when all are already inactive`() {
```

Comment:
This test case made me wonder if the endpoint should return all prison incentive levels or only the updated/deactivated ones.

Especially as the `incentives.prison-level.changed` domain event is published even for levels which were already inactive and didn't change (if I understand the current logic correctly).

Don't know if it would be better to only return the prison levels which were active/saved instead? Don't know if you thought about this already.

Reply:
**Update**: Discussed and decided that:
- domain events will be published only for prison levels that actually changed
- endpoint will return the complete list of levels for an enstiblishment